File: field/genLoads.py
"""
genLoads.py

Generate ARF excitation field for the QIBA curvilinear array for different focal
depths and attenuations.

Mark Palmeri
mlp6@duke.edu
2015-03-17
"""

import os
import shutil

# define some stuff
Freq = [3.0]
Fnum = [2.0, 3.5]
FD = [0.070, 0.050, 0.030]  # done in reverse order for normalization case to
                            # be generated first
alpha = [0.45]

root = '/getlab/mlp6/scratch/QIBA-DigitalPhantoms/field'

SLURM_TEMPLATE = 'genLoads.sh'

for i in range(len(Freq)):
    for j in range(len(Fnum)):
        for k in range(len(FD)):
            for l in range(len(alpha)):
                datafile = '%s/dyna-I-f%.2f-F%.1f-FD%.3f-a%.2f.mat' % \
                           (root, Freq[i], Fnum[j], FD[k], alpha[l])
                if not os.path.exists(datafile):
                    SLURM_FILENAME = datafile.replace('dyna-I-', 'field_')
                    SLURM_FILENAME = SLURM_FILENAME.replace('.mat', '.sh')
                    print SLURM_FILENAME
                    shutil.copy(SLURM_TEMPLATE, SLURM_FILENAME)
                    PARAM_STRING = '%.1f, %.3f, %.2f, %.2f' % \
                                   (Fnum[j], FD[k], Freq[i], alpha[l])
                    os.system('sed -i -e "s/PARAM_STRING/%s/" %s' %
                              (PARAM_STRING, SLURM_FILENAME))
                    os.system('sbatch %s' % (SLURM_FILENAME))
                else:
                    print('%s ALREADY EXISTS!!' % datafile)

File: field/genLoads.m
function genLoads(FNUM, FOCDEPTH, FREQ, ALPHA)
% function genLoads(FNUM, FOCDEPTH, FREQ, ALPHA)
%
% Call FEM/field Field II functions to generate ARF loads.
%
% Mark Palmeri
% mlp6@duke.edu
% 2015-03-19

FDmm = FOCDEPTH * 1000;

nodeFile = sprintf('../mesh/nodesFoc%.fmm.dyn', FDmm);

inputIntensityFile = field2dyna(nodeFile, ALPHA, FNUM, [0 0 FOCDEPTH], FREQ, ...
                                'curvlin_qiba', 'gaussian');

normIntensityFile = 'dyna-I-f3.00-F2.0-FD0.070-a0.45.mat';

makeLoadsTemps(inputIntensityFile, ...
               normIntensityFile, ...
               1000, 500, 2.0, 0.01^3, 'q', 1);

% generate VTU files for paraview
system(sprintf('python ~/git/fem/mesh/create_pointloads_vtk.py --nodefile ../mesh/nodesFoc%imm.dyn --loadfile PointLoads-f3.00-F%.1f-FD0.0%i-a0.45.dyn --loadout nodeLoadsFoc%immF%.1f', FDmm, FNUM, FDmm, FDmm, FNUM));

quit
